src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -270,17 +273,31 @@ public function isClosure() : bool
         return $this->node instanceof Node\Expr\Closure;
     }
 
-    /**
-     * Is this function deprecated?
-     *
-     * Note - we cannot reflect on internal functions (as there is no PHP source
-     * code we can access. This means, at present, we can only EVER return false
-     * from this function.
-     *
-     * @see https://github.com/Roave/BetterReflection/issues/38
-     */
     public function isDeprecated() : bool
     {
+        $docComment = $this->getDocComment();
+        if ($docComment !== '') {
+            $docBlockFactory = DocBlockFactory::createInstance();
+            $docBlock = $docBlockFactory->create($docComment);
+            if ($docBlock->hasTag('deprecated')) {
+                return true;
+            }
+        }
+
+        foreach ($this->getAttributes() as $attribute) {
+            if ($attribute->getName() === 'JetBrains\PhpStorm\Deprecated') {
+                $attributeArgs = $attribute->getArguments();
+                if (isset($attributeArgs['since'])) {
+                    // Convert version string to PHP_VERSION_ID equivalent.
+                    [$phpMajor, $phpMinor] = explode('.', $attributeArgs['since']);
+                    $phpVersionId = sprintf('%02d%02d00', $phpMajor, $phpMinor)
```

Comment:
There's already a place in the source code that does this: https://github.com/ondrejmirtes/BetterReflection/blob/348a8777d11b9dd88f13e0d99a84d1671a8cf8a7/src/SourceLocator/SourceStubber/PhpStormStubsSourceStubber.php#L399-L408

Please DRY and unify that in some static class helper (to the way that's in PhpStormStubsSourceStubber). Thanks.

Reply:
@ondrejmirtes gave a shot, moved it into static method `\PHPStan\BetterReflection\Util\PhpIdParser::fromVersion` which takes the semver value and converts to a PHP version ID. Naming was hard, went for something quick for your feedback.